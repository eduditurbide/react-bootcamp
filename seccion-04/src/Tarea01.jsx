// function App(params) {
//     // document.createElement ...
//     return (<h1>Hola Mundo!!!</h1>)
// }

// export default App

import React from 'react'

export const Tarea01 = () => {
  return (
    <h1>Tarea 01</h1>
  )
}
