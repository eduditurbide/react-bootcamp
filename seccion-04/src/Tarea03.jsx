import React from 'react'

export const Tarea03 = () => {
  return (
    <>
      <h1>Tarea 03</h1>
      <h3>Subtitulo</h3>
    </>
  )
}
