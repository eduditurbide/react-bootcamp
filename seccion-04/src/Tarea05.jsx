import React from 'react'
import PropTypes from 'prop-types'

export const Tarea05 = ({ title, subtitle }) => {
  return (
    <>
      <h1>{ title }</h1>
      <h3>{ subtitle }</h3>
    </>
  )
}

Tarea05.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
}