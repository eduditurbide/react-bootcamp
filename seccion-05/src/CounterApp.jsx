import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState( value )

  const handleAdd = () => setCounter( counter + 1 )
  const handleSubstract = () => setCounter( counter - 1 )
  const handleReset = () => setCounter( value )

  return (
    <>
      <h1>Tarea 07 - Counter App</h1>
      <h3>{ counter }</h3>

      <button onClick={ handleAdd }>+1</button>
      <button onClick={ handleSubstract }>-1</button>
      <button aria-label='btn-reset' onClick={ handleReset }>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}