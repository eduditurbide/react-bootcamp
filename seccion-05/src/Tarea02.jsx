import React from 'react'

const newMessage = {
    message: "Nuevo mensaje",
    title: "Nuevo título"
}

const getResult = (a, b) => {
  return a + b
}

export const Tarea02 = () => {

  return (
    <>
        <h1>Tarea 02</h1>
        <p>Soy un subtitulo</p>

        <code>{ JSON.stringify(newMessage) }</code>

        <p>Suma de 1 más 2 = { getResult(1,2) }</p>

    </>
  )
}
