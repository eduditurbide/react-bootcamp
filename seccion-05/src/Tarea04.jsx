import React from 'react'

export const Tarea04 = ({ title = "Tarea 04 por default", subtitle }) => {
  return (
    <>
      <h1>{ title }</h1>
      <h3>{ subtitle }</h3>
    </>
  )
}
