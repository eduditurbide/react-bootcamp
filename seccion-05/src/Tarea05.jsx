import React from 'react'
import PropTypes from 'prop-types'

export const Tarea05 = ({ title, subtitle }) => {
  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      <h3>{ subtitle }</h3>

      <p>Un subtitulo por props</p>
    </>
  )
}

Tarea05.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
}