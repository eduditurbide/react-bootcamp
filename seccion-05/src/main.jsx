import React from 'react'
import ReactDOM from 'react-dom/client'
import './style.css'
import { Tarea01 } from './Tarea01'
import { Tarea02 } from './Tarea02'
import { Tarea03 } from './Tarea03'
import { Tarea04 } from './Tarea04'
import { Tarea05 } from './Tarea05'
import { Tarea06 } from './Tarea06'
import { CounterApp } from './CounterApp'

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <Tarea01 /> <hr/>
        <Tarea02 /> <hr/>
        <Tarea03 /> <hr/>
        <Tarea04 subtitle="Un subtitulo por props"/> <hr/>
        <Tarea05 title='Tarea 05' subtitle={1234}/> <hr/>
        <Tarea06 /> <hr/>
        <CounterApp value={34}/> <hr/>
    </React.StrictMode>
)