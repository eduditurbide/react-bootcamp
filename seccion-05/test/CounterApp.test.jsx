import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Test <CounterApp />', () => {

  const valorInicial = 100
  
  test('debe ser match con el snapshot', () => { 
   
    const { container, getByText } = render( <CounterApp value={ valorInicial }/>  )
    expect( container ).toMatchSnapshot()

  })

  test('debe de mostrar el valor inicial de 100', () => { 
   
    render( <CounterApp value={ valorInicial }/>  )
    expect( screen.getByText(100) ).toBeTruthy()
    expect( screen.getByRole('heading', { level: 3}).innerHTML ).toContain(`${valorInicial}`)

  })

  test('debe de incrementar con el botón +1', () => { 
    
    render( <CounterApp value={ valorInicial }/>  )
    fireEvent.click( screen.getByText('+1') )
    expect( screen.getByText(101) ).toBeTruthy()

  })

  test('debe de decrementar con el botón -1', () => { 
    
    render( <CounterApp value={ valorInicial }/>  )
    fireEvent.click( screen.getByText('-1') )
    expect( screen.getByText(99) ).toBeTruthy()

  })

  test('debe de volver al valor inicial con el botón reset', () => { 
    
    render( <CounterApp value={ valorInicial }/>  )
    fireEvent.click( screen.getByText('+1') )
    fireEvent.click( screen.getByText('+1') )
    fireEvent.click( screen.getByText('+1') )

    // fireEvent.click( screen.getByText('Reset') )
    // expect( screen.getByText(valorInicial) ).toBeTruthy()
    
    fireEvent.click( screen.getByRole('button', {name: 'btn-reset'}) )
    expect( screen.getByText(valorInicial) ).toBeTruthy()

  })
  
})
