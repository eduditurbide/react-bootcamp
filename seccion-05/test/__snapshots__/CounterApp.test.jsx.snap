// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <CounterApp /> debe ser match con el snapshot 1`] = `
<div>
  <h1>
    Tarea 07 - Counter App
  </h1>
  <h3>
    100
  </h3>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
